envoy/api/v2/route/route.proto

```diff
@@ -803,7 +814,36 @@ message HeaderMatcher {
   // * The regex *\d{3}* matches the value *123*
   // * The regex *\d{3}* does not match the value *1234*
   // * The regex *\d{3}* does not match the value *123.456*
-  google.protobuf.BoolValue regex = 3;
+  google.protobuf.BoolValue regex = 3 [deprecated = true];
+
+  // [#not-implemented-hide:]
+  // Specifies how the header match will be performed to route the request.
+  // If header_match_specifier is absent, a request that has the name header will match, regardless
```

Comment:
"a request that has :ref:`header <internal link back to header field>` will match". I know this problem was there before, this is a good place to cleanup.